PipelinesTasks/PRMetrics/buildAndReleaseTask/invokers/azureReposInvoker.ts

```diff
@@ -0,0 +1,219 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { IGitApi } from "azure-devops-node-api/gitApi";
+import { JsonPatchDocument, JsonPatchOperation, Operation } from "azure-devops-node-api/interfaces/common/VSSInterfaces";
+import { Comment, CommentThreadStatus, GitPullRequest, GitPullRequestCommentThread, GitPullRequestIteration } from "azure-devops-node-api/interfaces/GitInterfaces";
+import DevOpsApiWrapper from "../wrappers/devOpsApiWrapper";
+import TaskLibWrapper from "../wrappers/taskLibWrapper";
+
+export default class AzureReposInvoker {
+    private taskLibWrapper: TaskLibWrapper;
+    private devOpsApiWrapper: DevOpsApiWrapper;
+    private gitApi: IGitApi | undefined;
+
+    private baseUri = `${process.env["SYSTEM_TEAMFOUNDATIONCOLLECTIONURI"]}`;
+    private project = process.env["SYSTEM_TEAMPROJECT"];
+    private repositoryId = process.env["BUILD_REPOSITORY_ID"] as string;
+    private pullRequestId = process.env["SYSTEM_PULLREQUEST_PULLREQUESTID"] ? parseInt(process.env["SYSTEM_PULLREQUEST_PULLREQUESTID"]) : 0;
+    private azurePAT = process.env["SYSTEM_ACCESSTOKEN"];
+
+    /**
+      * Initializes a new instance of the AzureReposInvoker class.
+      * @param devOpsApiWrapper The wrapper around the Azure Devops Api Task Lib.
+      * @param taskLibWrapper The wrapper around the Azure Pipelines Task Lib.
+      */
+    public constructor(devOpsApiWrapper: DevOpsApiWrapper, taskLibWrapper: TaskLibWrapper) {
+        this.devOpsApiWrapper = devOpsApiWrapper;
+        this.taskLibWrapper = taskLibWrapper;
+    }
+
+    /**
+      * Gets the pull request from the devops api.
+      */
+    public getDetails = async (): Promise<GitPullRequest> => {
+        this.taskLibWrapper.debug('* AzureReposInvoker.getDetails()');
+
+        const gitApi = await this.openConnection();
+        return await gitApi.getPullRequestById(this.pullRequestId, this.project);
+    }
+
+    /**
+      * Gets the pull request iterations from the devops api.
+      */
+    public getIterations = async (): Promise<GitPullRequestIteration[]> => {
+        this.taskLibWrapper.debug('* AzureReposInvoker.getIterations()');
+
+        const gitApi = await this.openConnection();
+        return await gitApi.getPullRequestIterations(this.repositoryId, this.pullRequestId, this.project);
+    }
+
+    /**
+      * Gets the pull request comment threads from the devops api.
+      */
+    public getCommentThreads = async (): Promise<GitPullRequestCommentThread[]> => {
```

Comment:
The comment threads are to get comments in:
- files marked as not requiring updates
- the stats comment

I don't think this can be easily done here, so we'll likely need to return all of this info I guess...